app/helpers/application_helper.rb

```diff
@@ -1,2 +1,7 @@
 module ApplicationHelper
+  @@renderer = Redcarpet::Markdown.new(Redcarpet::Render::HTML, autolink: true, tables: true)
```

Comment:
Good question! 

Class variables can be useful for initialization data. In this case, I'm defining a class variable on a module. By doing that, I ensure that the "html renderer" will be instantiated only once per rails app, as that class variable (`@@renderer`) of the `ApplicationHelper` module will be *shared* among all classes which will include that module (namely, all controllers and all views). In effect, that class variable of that `ApplicationHelper` module will also become a class variable of any class including that `ApplicationHelper` module.

Proof:

```ruby
module MyModule
  @@var = 0
  class << self
    def var
      @@var
    end
  end

  def increment_var
    @@var += 1
  end
end

class ClassA
  include MyModule
end

class ClassB
  include MyModule
end

puts MyModule.class_variables.inspect
puts MyModule.var

ClassA.new.increment_var
puts MyModule.var

ClassB.new.increment_var
puts MyModule.var
```

will output:

```ruby
[:@@var]
0
1
2
```

And to answer your question more directly, if the renderer would be defined in the instance method of the module, then a new renderer would be instantiated each time that method would be called. So defining the renderer as a class variable is a matter of both performance and memory efficiency.

Note that using class variables for class state (as opposed to initialization data) can be a bad idea. [This article](http://www.sitepoint.com/class-variables-a-ruby-gotcha/) gives an example of this, if you're curious.